staubli_val3_driver/val3/ros_server/decTrajPt.pgx

```diff
@@ -32,30 +32,43 @@
 
   // sequence
   l_nRetVal=fromBinary(rosTrajPtMsg.body.nData,4,"-4l",rosTrajPtMsg.jointTrajPt.nSeq)
-  if (l_nRetVal!=1)
+  if(l_nRetVal!=1)
     x_bFlag=false
     return
   endIf
 
   // joints[10]
   for l_nIndex=0 to 9 step 1
     l_nRetVal=fromBinary(rosTrajPtMsg.body.nData[(l_nIndex*4)+4],4,"4.0l",rosTrajPtMsg.jointTrajPt.nJoints[l_nIndex])
-    if (l_nRetVal!=1)
+    if(l_nRetVal!=1)
       x_bFlag=false
       return
     endIf
   endFor
 
+ // Velocity buffer to smooth out last traj point
+  nVel[1] = nVel[0]
+  nVel[0] = rosTrajPtMsg.jointTrajPt.nVelocity
+
   // velocity
   l_nRetVal=fromBinary(rosTrajPtMsg.body.nData[44],4,"4.0l",rosTrajPtMsg.jointTrajPt.nVelocity)
-  if (l_nRetVal!=1)
+  if(l_nRetVal!=1)
     x_bFlag=false
     return
   endIf
 
+ // Round to remove trailing decimal places from fromBinary()
+  rosTrajPtMsg.jointTrajPt.nVelocity = rosTrajPtMsg.jointTrajPt.nVelocity * 10000
+  rosTrajPtMsg.jointTrajPt.nVelocity = round(rosTrajPtMsg.jointTrajPt.nVelocity)
+  rosTrajPtMsg.jointTrajPt.nVelocity = rosTrajPtMsg.jointTrajPt.nVelocity / 10000
+  // Smooth last point in traj
+  if((rosTrajPtMsg.jointTrajPt.nVelocity == 0.1) and (rosTrajPtMsg.jointTrajPt.nSeq != 0))
```

Comment:
Suggestion: add a comment explaining why you check `0.1` and `nSeq != 0`.

Reply:
👍